pyhanko/cli.py

```diff
@@ -471,14 +476,15 @@ def validate_signatures(ctx, infile, executive_summary,
     )
     with pyhanko_exception_manager():
         if detached is not None:
-            status_str = _signature_status_str(
+            (status_str, signature_ok) = _signature_status_str(
                 status_callback=lambda: _validate_detached(
                     infile, detached, ValidationContext(**vc_kwargs),
                     key_usage_settings
                 ),
                 pretty_print=pretty_print, executive_summary=executive_summary
             )
             print(status_str)
+            exit_if_invalid(signature_ok)
```

Comment:
Could you replace this with something like the following?

```
if not signature_ok:
     raise click.ClickException(....error message goes here...)
```

I prefer this construct for two reasons:

- It allows Click to gracefully handle whatever it needs to handle, and should cause the program to exit with exit code 1. The error message is printed on `stderr`.
- It's obvious from the syntax that the control flow is being interrupted here. `exit_if_invalid(...)` looks like an ordinary function call (other than the name), which is visually a bit jarring and tends to confuse IDEs.

If you need another exit code, you can set the `exit_code` property on the exception object. That said, I don't think any of the exit codes with conventionally understood meanings are appropriate here, so I propose we stick to exit status 1 for now (but I'm willing to have my mind changed on this one).

Scripts that need more detailed information can always parse `stdout`. PyHanko's non-prettyprint output is intended to be consumed like that :).

Reply:
Done!